tests/SubTests.fs

```diff
@@ -0,0 +1,95 @@
+﻿module Elmish.SubTests
+
+open Elmish
+open NUnit.Framework
+open System
+
+// TIL: Each use of a let fn binding creates a new FSharpFunc object.
+// Meaning reference equality with let fn binding is always false.
+// As record properties, these become concrete objects, ref equality works.
+type SubContainer =
+    { Sub: SubId -> Dispatch<obj> -> IDisposable
+      Dupe: SubId -> Dispatch<obj> -> IDisposable }
+
+[<TestFixture>]
+type DiffBehavior() =
+    // data
+    let stop = {new IDisposable with member _.Dispose() = () }
+    let sub = {Sub = (fun _ _  -> stop); Dupe = (fun _ _ -> stop) }
+    let newId i = sprintf "sub/%i" i
+    let gen idRangeStart idRangeEnd second =
+        let count = idRangeEnd + 1 - idRangeStart
+        List.init count (fun i -> newId (idRangeStart + i), second)
+    let genSub idRangeStart idRangeEnd whichSub =
+        let count = idRangeEnd + 1 - idRangeStart
+        List.init count (fun i ->
+            { SubId = newId (idRangeStart + i)
+              Start = whichSub })
+    // helpers
+    let toKeys keyValueList =
+        List.map fst keyValueList
+    let toKeys2 subs =
+        subs |> List.map (fun sub -> sub.SubId)
+    let toIds (dupes, toStop, toKeep, toStart) =
+        dupes, toKeys toStop, toKeys toKeep, toKeys toStart
+    let toIds2 (dupes, toStop, toKeep, toStart) =
+        toKeys2 dupes, toKeys toStop, toKeys toKeep, toKeys2 toStart
+    let run = Sub.Internal.diff
+    let eq expected actual (message: string) =
```

Comment:
Unquote lib provides nice `=!` operator that reports expected and actual, if equality assertion fails.

Reply:
I ended up keeping eq to avoid repeating `toId` fns in every test. But it now uses `=!` so the data can be seen on failure.